### PR TITLE
Readjusting the version numbers

### PR DESCRIPTION
**Pull request by sburmanoctopus**
[sc-62820]

Updating the version numbers in our docs to be up-to-date for the RPC retries and Event Retention features.

### CODE REVIEW

File: src/pages/docs/infrastructure/deployment-targets/machine-policies.mdx [hunk 1]
```diff
@@ -153,7 +153,7 @@ You can select a username/password account to perform automatic Tentacle updates
 ## Recover from communication errors with Tentacle \{#MachinePolicies-Recoverfromcommunicationerrorswithtentacle}
 
 :::div{.hint}
-Recovering from communication errors with Tentacle is available to **Cloud** customers from **Octopus 2023.3** onwards and will be made available to self-hosted customers in a future release.
+Recovering from communication errors with Tentacle is available from **Octopus 2023.4** onwards.
```
Comment (sburmanoctopus) [inline src/pages/docs/infrastructure/deployment-targets/machine-policies.mdx:156]: Now that we know 2023.4 is the version, let's lock that in.

Also, everyone in Cloud has RPC retries now. So it felt strange to distinguish them here. If a Cloud customer looks at their version, they will see it's >= 2023.4.

File: src/pages/docs/security/users-and-teams/auditing/index.md [hunk 2]
```diff
@@ -66,9 +66,7 @@ We take the sensitive value and hash it using an irreversible hash algorithm. We
 ## Archived audit logs {#archived-audit-events}
 
 :::div{.hint}
-The audit log retention functionality was gradually made available to **Cloud** customers from **Octopus 2023.1** onwards.
-
-Audit log retention will be made available to self-hosted customers later in 2023. However, from **Octopus 2022.3** onwards, self-hosted customers will be able to change the **Event retention days** configuration, allowing you to adjust to your preferred retention period prior to retention archiving audit logs. 
+The audit log retention functionality is available from **Octopus 2023.3** onwards.
```
Comment (sburmanoctopus) [inline src/pages/docs/security/users-and-teams/auditing/index.md:69]: Having done the fix to RPC retries, I was reminded that we do similar version distinctions in event retention.

Applying the same logic, I felt that it has been around for a while now, and didn't need as much fan-fair now.

Comment (sburmanoctopus) [reply src/pages/docs/security/users-and-teams/auditing/index.md:69]: I'm pretty sure this became available to the public in `2023.3`. [The feature toggle was removed then](https://github.com/OctopusDeploy/OctopusDeploy/commit/229ab39e085096650991e14f947506e983360b43), so I believe that's good evidence 😁

File: src/pages/docs/infrastructure/deployment-targets/machine-policies.mdx (hunk 1 above)
Comment (shipra-octopus) [reply src/pages/docs/infrastructure/deployment-targets/machine-policies.mdx:156]: LGTM

File: src/pages/docs/security/users-and-teams/auditing/index.md (hunk 2 above)
Comment (shipra-octopus) [reply src/pages/docs/security/users-and-teams/auditing/index.md:69]: LGTM

**Comment by shipra-octopus** [comment]
Lood good Stephen! Thank you!

**Comment by shipra-octopus** [comment]
Lood good Stephen! Thank you!

**Review by shipra-octopus — APPROVED** [review]
Looks good Stephen! Thank you!

**Pull request merged**